login.html

```diff
@@ -130,14 +130,18 @@
     data.append('password', password.value);
     xhr = new XMLHttpRequest();
     xhr.onreadystatechange = function() {
-      if (xhr.readyState == 4 && xhr.status==200) {
-        result = JSON.parse(xhr.responseText);
-        if (result.error == "") {
-          successFunction(result);
+      if (xhr.readyState == 4) {
+        if (xhr.status==200) {
+          result = JSON.parse(xhr.responseText);
+          if (result.error == "") {
+            successFunction(result);
+          } else {
+            errorFunction(result);
+          }
+          locked = false;
         } else {
```

Comment:
I think you should call errorFunction with a UNKNOWN_ERROR constant, e.g.

var UNKNOWN_ERROR = { 'error': 'n o worky' } 

at file scope